internalexecutor fails to handle panic correctly

**Describe the problem**

If internal executor hits panic when processing query, it will fail handling it.

**To Reproduce**

Within server startup call:

```
_, _ = s.node.sqlExec.Exec(ctx, "", nil, "select * from crdb_internal.rangelog where range_id = $1", []interface{}{})
```
to cause assertion panic on query argument types.

Stderr will contain:

```
I220105 12:34:05.201663 1 util/log/flags.go:202  [-] 1  stderr capture started
panic: unexpected type []interface {}
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6aa8652]

goroutine 87 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal.func1()
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:676 +0x192
panic({0x80cc2c0, 0xc000752700})
        /usr/local/opt/go/libexec/src/runtime/panic.go:1038 +0x215
github.com/cockroachdb/cockroach/pkg/sql.golangFillQueryArguments({0xc000ae7e58, 0x1, 0x0})
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/sql/exec_util.go:1598 +0xb1a
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal(0xc000481b60, {0xbbf5828, 0xc0004d8090}, {0x0, 0x0}, 0xc00107b380, 0x48, {{{0x84ecec7, 0x4}}, {0x0, ...}, ...}, ...)
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:701 +0x568
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).ExecEx(0x2, {0xbbf5828, 0xc0004d8090}, {0x0, 0x0}, 0x8, {{{0x84ecec7, 0x4}}, {0x0, 0x0}, ...}, ...)
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:529 +0x18e
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).Exec(0x12, {0xbbf5828, 0xc0004d8090}, {0x0, 0x0}, 0x0, {0x861b7e6, 0x38}, {0xc000ae7e58, 0x1, ...})
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:510 +0x150
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc000c75200, {0xbbf5828, 0xc0004d8090})
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1946 +0x341f
github.com/cockroachdb/cockroach/pkg/cli.runStart.func4.2(0xc000d66750, 0xc000486fa0, 0xc000790f70, {0xbbf5828, 0xc0004d8090}, 0x1, {0xbf0dab0, 0xed967873d, 0x0})
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:648 +0x165
github.com/cockroachdb/cockroach/pkg/cli.runStart.func4()
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:688 +0xf6
created by github.com/cockroachdb/cockroach/pkg/cli.runStart
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:604 +0x945
```

**Expected behavior**
Logs should report proper source of panic:

```
I220105 12:26:46.521099 1 util/log/flags.go:202  [-] 1  stderr capture started
panic: unexpected type []interface {}

goroutine 109 [running]:
github.com/cockroachdb/cockroach/pkg/sql.golangFillQueryArguments({0xc001fafe58, 0x1, 0x0})
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/sql/exec_util.go:1598 +0xb1a
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal(0xc0010b8660, {0xbbf5808, 0xc001140c00}, {0x0, 0x0}, 0xc0011a0e00, 0x48, {{{0x84ecea7, 0x4}}, {0x0, ...}, ...}, ...)
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:700 +0x568
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).ExecEx(0x2, {0xbbf5808, 0xc001140c00}, {0x0, 0x0}, 0x8, {{{0x84ecea7, 0x4}}, {0x0, 0x0}, ...}, ...)
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:529 +0x18e
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).Exec(0x12, {0xbbf5808, 0xc001140c00}, {0x0, 0x0}, 0x0, {0x861b7c6, 0x38}, {0xc001fafe58, 0x1, ...})
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:510 +0x150
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc000d0e900, {0xbbf5808, 0xc001140c00})
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1946 +0x341f
github.com/cockroachdb/cockroach/pkg/cli.runStart.func4.2(0xc00053d560, 0xc00017cf00, 0xc000b8a160, {0xbbf5808, 0xc001140c00}, 0x1, {0x1efbee38, 0xed9678586, 0x0})
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:648 +0x165
github.com/cockroachdb/cockroach/pkg/cli.runStart.func4()
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:688 +0xf6
created by github.com/cockroachdb/cockroach/pkg/cli.runStart
        /Users/xxx/go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:604 +0x945
```

**Additional context**
There's no production impact as assertions are triggered by faulty server side code. It is helpful though to have correct errors for engineer comfort.

--- a/pkg/sql/internal.go
+++ b/pkg/sql/internal.go
@@ -662,15 +662,15 @@ func (ie *InternalExecutor) execInternal(
 		//
 		// TODO(knz): track the callers and check whether opName could be turned
 		// into a type safe for reporting.
-		if retErr != nil {
-			if !errIsRetriable(retErr) {
+		if retErr != nil || r == nil {
+			// Both retErr and r can be nil in case of panic.
+			if retErr != nil && !errIsRetriable(retErr) {
 				retErr = errors.Wrapf(retErr, "%s", opName)
 			}
 			stmtBuf.Close()
 			wg.Wait()
 			sp.Finish()
 		} else {
-			// r must be non-nil here.
 			r.errCallback = func(err error) error {
 				if err != nil && !errIsRetriable(err) {
 					err = errors.Wrapf(err, "%s", opName)

